### ISSUE TITLE
[BUG] Lombok is not Performing Injections in Custom Gradle Task

### ISSUE THREAD

**Issue by MorningSage** [rzwitserloot/lombok#2665]
**Describe the bug**
The annotation processor appears to be ignored when using custom gradle tasks.

**To Reproduce**
Using IntelliJ and any version of Lombok >1.18.x, create a custom gradle task before running the main one.  IntelliJ does not complain, but the compiler does not see any of Lombok's changes.

Here is a series of images in case what I'm saying doesn't make sense:
https://imgur.com/a/tgj0Sqc

**Expected behavior**
The expected behavior is for Lombok to correctly function as an annotation processor when running any task in Gradle.

**Version info (please complete the following information):**
 - Lombok version: >1.18.x
 - Platform (javac or eclipse, and if so, what is the output of `javac -version` / the version listed in the _about..._ dialog of eclipse.
   - IntelliJ IDEA 2020.2.3 (Community Edition)
Build #IC-202.7660.26, built on October 6, 2020
   - C:\Program Files\Java\jdk1.8.0_191\bin>.\javac.exe -version
javac 1.8.0_191

**Additional context**
Using Lombok version 1.16.22 works well.  Whatever caused this to break was introduced in 1.18.x.  If there's anything else I can provide, please let me know.  This appears to be similar to https://github.com/gradle/gradle/issues/10997 but I don't know what I'm talking about.

**Comment by Rawi01** [comment]
I don't think that there is a relevant change between [1.16.22 and 1.18.0](https://github.com/rzwitserloot/lombok/compare/v1.16.22...v1.18.0). Have you tried to run your task without InteliJ?

**Comment by MorningSage** [comment]
I hadn't actually, but I still get the same error running `gradlew generateData` directly with IntelliJ closed.
